kuma/wiki/models.py

```diff
@@ -1528,8 +1535,10 @@ def clean_current_revision(self, user):
         # Populate the model instance with fresh data from database.
         rev.refresh_from_db()
 
-        # Make this new revision the current one for the document.
-        rev.make_current()
+        # Make this new revision the current one for the document if it hasn't
+        # been done already when saving the revision above.
+        if not rev.is_approved:
```

Comment:
Is it possible to write `if self.current_revision is not rev:` here instead?  (I'm not sure what the right way to compare is... maybe comparing the pk fields of the revisions?)

I've got no idea what the approval status of the revision has to do with the behavior of rev.save(). So if it is possible to test for this condition without requiring internal knowledge of how save() works, I'd prefer that. But if this is the best way to do it, then I'm fine with this as an optimization.

Also, I'm assuming that if the atomic transaction above fails, that an error is raised and we never get here. Right?

Reply:
You're absolutely right, calling `rev.make_current()` based on `rev.is_approved` requires internal knowledge of `save()`, and something like `if self.current_revision.pk is not rev.pk` is better, thanks!

Reply:
Oh, and in terms of what happens when an error occurs within the atomic transaction, yes, the error will be raised and the rest of the code will never be executed.